### DISCUSSION TITLE
`iframe` compatibility issue between plugins

### DISCUSSION THREAD

**Discussion by Nriver** [TriliumNext/Trilium#4290] [General]
I've found that https://github.com/SiriusXT/trilium-drawio and https://github.com/rauenzi/Trilium-SingleFile are not compatible at this point. It may have something to do with the CSS for the iframe, as both of them are using iframes.

With both of them installed, Drawio works fine.
![ksnip_20230925-113948](https://github.com/zadam/trilium/assets/6752679/3bb3b957-d324-4fb2-ab71-ddb23ae3ee6a)

However, it makes the page clipped by SingleFile like this, and the height does not look right:
![ksnip_20230925-113156](https://github.com/zadam/trilium/assets/6752679/c48aa61b-2c33-4fb7-89a2-bb2ebb98d657)

By removing these lines from trilium-drawio, the page for SingleFile will be fine.
[Link to the lines to be removed](https://github.com/SiriusXT/trilium-drawio/blob/0f642c3110477200cb769a24f1a7ae56e8e0e38a/trilium-drawio.js#L198C1-L204C4)
![ksnip_20230925-113613](https://github.com/zadam/trilium/assets/6752679/8702a587-6b54-4a19-b84c-415c63f5f90f)

But it will break Drawio.
![ksnip_20230925-113640](https://github.com/zadam/trilium/assets/6752679/6ac103c4-acda-470c-884d-a2f151e2df93)

😀 What should we do?
@SiriusXT @rauenzi

**Comment by Nriver** [comment]
The follow-up question is: if there is another plugin using iframe, a similar issue could arise. What action should we take to prevent this?

**Reply by zerebos** [reply]
You can modify the erroneous plugin to use a more specific selector or request the dev do so.

**Comment by zxccqxue** [comment]
unsubscribe

Nriver ***@***.***> 于2023年9月25日周一 11:49写道：

**Comment by zerebos** [comment]
The problem here is the `drawio` plugin is using too global a selector which affects any and all `<iframe>`s on the page. Addons should try to limit their css selectors to only be affecting what they intend rather than general effects.

Unfortunately there's no real way to [scope plain css](https://caniuse.com/?search=scoped). So developers will either need to add specificity to their css selectors to ensure they're only touching their own elements, or they need to make use of [Web Components](https://developer.mozilla.org/en-US/docs/Web/API/Web_components) which can scope styles, but it is not nearly as easy or straightforward as adding plain css and elements to the page.
